yt/frontends/gadget/fields.py

```diff
@@ -122,6 +128,22 @@ def _h_density(field, data):
 
         return ["H"] + metal_names[:-1]
 
+    def _check_whitelist_species_fields(self, ptype):
+        species_names = []
+        species_nums = []
+        for field in self.ds.field_list:
+            if field[0] == ptype:
+                if field[1].endswith("_fraction") or field[1].endswith("_density"):
+                    if field[1] in sph_whitelist_fields:
+                        symbol = field[1].split("_")[0]
+                        species_names.append(symbol)
+                        species_nums.append(periodic_table[symbol].num)
+        if len(species_names) > 0:
+            ret = list(sort_together([species_nums, species_names])[1])
```

Comment:
Can you please explain why we need to select just the `[1]` element ?

Reply:
@neutrinoceros what we are trying to do here is get a list of elements in the dataset--but we need the list of elements to be sorted by atomic number, so I `sort_together` the two lists of atomic numbers and names but then just discard the former.